releasenotes/notes/improve-performance-merge-adjacent-barriers-pass-c0cd064bdd059811.yaml

```
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    Performance improvement of the :class:`~.MergeAdjacentBarriers` transpiler pass, which used to
+    rebuild the complete DAG to merge the barriers. It now iterates over the barriers instead of
+    the full graph, and applies the :meth:`.`DAGCircuit.replace_block_with_op` method when necessary.
```

```suggestion
    the full graph, and applies the :meth:`.DAGCircuit.replace_block_with_op` method when necessary.
```

But more likely: we probably don't need to say the last phrase at all, because it's kind of an internal implementation detail that users don't need to know.